lib/api.es6.js

```diff
@@ -50,6 +47,102 @@ const login = (apiOptions, username, pass) => {
   });
 };
 
+const register = (apiOptions, username, password, email, newsletter) => {
+  return new Promise((r, x) => {
+    if (!apiOptions.oauthAppOrigin) {
+      x('Please set up a Reddit Oauth App, and pass in its URL as oauthAppOrigin to config.');
+    }
+
+    if (!apiOptions.clientId) {
+      x('Please set up a Reddit Oauth App, and pass in its id as clientId to config.');
+    }
+
+    if (!apiOptions.clientSecret) {
+      x('Please set up a Reddit Oauth App, and pass in its secret as clientSecret to config.');
+    }
+
+    const data = {
+      user: username,
+      passwd: password,
+      passwd2: password,
+      api_type: 'json',
+    };
+
+    if (email) {
+      data.email = email;
+    }
+
+    if (newsletter) {
+      data.newsletter_subscribe = true;
+
+      if (!email) {
+        x('NEWSLETTER_NO_EMAIL');
```

Comment:
Is there a reason some of these rejection messages seem like "token" type messages while some seem more colloquial?